### ISSUE TITLE
Change docs to recommend TypeToken.getParameterized instead of anonymous classes

### ISSUE THREAD

**Issue by Marcono1234** [google/gson#1519]
Currently the documentation recommends creating an anonymous class for `TypeToken`s of generic classes. Some examples (there might be more):
https://github.com/google/gson/blob/63ee47cb642c8018e5cddd639aa2be143220ad4b/gson/src/main/java/com/google/gson/reflect/TypeToken.java#L38
https://github.com/google/gson/blob/63ee47cb642c8018e5cddd639aa2be143220ad4b/gson/src/main/java/com/google/gson/Gson.java#L87

#900 added the method `TypeToken.getParameterized` which should probably now be the preferred way of creating these `TypeToken`s.

Could the documentation please be updated?

**Comment by lyubomyr-shaydariv** [comment]
`javac`/etc can check (anonymous) classes at compile-time (i.e., number of type parameters, type parameter bounds), whilst `TypeToken.getParameterized(...)` cannot be checked unless executed. For example, this will fail:

```java
gson.fromJson("[1,2,3]", TypeToken.getParameterized(List.class).getType());
```

This will not fail but makes no sense (I believe `getCollectionElementType` does not check for exactly 1 type parameter, and should probably be fixed):

```java
gson.fromJson("[1,2,3]", TypeToken.getParameterized(List.class, Integer.class, Whatever.class).getType());
```

This will fail either (due to insufficient but a bit more efficient check):

```java
gson.fromJson("{\"foo\":1}", TypeToken.getParameterized(Map.class, String.class).getType());
```

I would prefer anonymous classes due to compile-time safety.

**Comment by Marcono1234** [comment]
Good point. But should it at least be mentioned as alternative? Or do you think the users for whom it might become relevant will find it on their own?
